docs/api.md

```diff
@@ -117,8 +117,10 @@ The format of the template of one row of Pipe 2 is as follows:
   "statement": {statementBlob},
   "itemId": [itemId],
   "reference": {
-    referenceMetadata: referenceBlob,
-    extractedData: [ [value] ]
+    "url": [referenceURL],
```

Comment:
What would you think about also putting the "date_retrieved" alongside here? Amir mentioned something yesterday that made sense to me that that date should also be stored near the URL?

Reply:
I think it would only make sense stored near the url when constructing the potential reference, but maybe @Ladsgroup has a different idea, that I'm not aware of.

Reply:
> > In light of gaining some more experience with the `referenceMetadata`, I'd like to propose a different approach here, as the current structure `<pid>: <value>` Is causing us not only to write unnecessary loops to extract the data we want, but also hardcode the `pid`s we are trying to extract in various places in the code. I would suggest that we switch to a different stricture as such:
> > ```js
> > { 
> >   // ...
> >   "referenceMetdata": {
> >      "statedIn": {
> >          "pid": String // pid for stated in (always 248 in our case),
> >          "value": String // value for stated in
> >       },
> >      "externalId": {
> >          "pid": String // pid external Id,
> >          "value": String // value for external id
> >       },
> >      "referenceUrl": {
> >          "pid": String // pid for reference URL (always 854 in our case),
> >          "value": String // value for ref URL
> >       },
> >      "dateRetrieved": {
> >          "pid": String // pid for date retrieved (always 813 in our case),
> >          "value": String // value for date retrieved
> >       }
> >   }
> > ```
> > 
> > 
> > }
> 
> I quite like the idea, my only worry right now is the backward compatibility in the code but it's not related to this case.

I also like this idea; my original thought about the referenceMetadata is that it would be most convenient if it was exactly the payload of the reference needed in the Wikidata api call.

This would mean that the api.php script could be super simple and not need to know anything about this data. It would just wrap it up and make the call.

However, as we've already found recently we might need to both include and exclude values in the references and we don't want to need to re-run the whole pipeline in order to be able to do that. Given that api.php therefore needs to be able to parse this referenceMetadata key and not just ship-it wholesale I think it makes sense to mark it up in this way.